torch/optim/optimizer.py

```diff
@@ -164,16 +164,33 @@ def update_group(group, new_group):
             update_group(g, ng) for g, ng in zip(groups, saved_groups)]
         self.__setstate__({'state': state, 'param_groups': param_groups})
 
-    def zero_grad(self):
-        r"""Clears the gradients of all optimized :class:`torch.Tensor` s."""
+    def zero_grad(self, set_to_none=False):
+        r"""Set the gradients of all optimized :class:`torch.Tensor` s to zero.
+
+        Arguments:
+            set_to_none (bool): instead of setting to zero, set the grad to None.
+                This will in general have lower memory footprint, but using this
+                comes with caveats, to name a few:
+                1. When user tries to access the gradient value and perform manual ops on it.
+                A None attribute or a Tensor full of 0s will be different.
+                2. User can no longer rely on checking if `.grad` is None to see if a tensor
```

Comment:
this doesn't make sense.  I'd say it's the opposite:  if a user sets grads = None before backward, they CAN check if `.grad` is None to see if the tensor received a gradient in the backward pass.  Let's just describe the behavior:
```
If the user requests `zero_grad(set_to_none=True)` followed by a backward pass, `.grad`\ s are guaranteed to be None for params that did not receive a gradient.
```